File: samples/dotnet-isolated-azurefunction/Trigger/ConsumeMessageFromKafka.cs
// ------------------------------------------------------------
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
// ------------------------------------------------------------

namespace dotnet_isolated_azurefunction
{
    using System.Text.Json;
    using Microsoft.Azure.Functions.Worker;
    using Microsoft.Azure.Functions.Worker.Extensions.Dapr;
    using Microsoft.Extensions.Logging;

    public static class ConsumeMessageFromKafka
    {
        // The function is triggered by Kafka messages in the Kafka instance referenced by
        // the Kafka binding configured under components/kafka-bindings.yaml
        // Can be used as an alternative for the node-app in the Dapr Bindings sample
        // found at https://github.com/dapr/quickstarts/tree/master/bindings/nodeapp
        [Function("ConsumeMessageFromKafka")]
        public static void Run(
            // Note: the value of BindingName must match the binding name in components/kafka-bindings.yaml
            [DaprBindingTrigger(BindingName = "%KafkaBindingName%")] JsonElement triggerData,
            FunctionContext functionContext)
        {
            var log = functionContext.GetLogger("ConsumeMessageFromKafka");

            log.LogInformation("Hello from Kafka!");

            log.LogInformation($"Trigger data: {triggerData}");
        }
    }
}

File: samples/python-v2-azurefunctions/consume_message_from_kafka.py
import logging 
import azure.functions as func 

consumeMessageFromKafka = func.DaprBlueprint()

# Dapr binding trigger
@consumeMessageFromKafka.function_name(name="ConsumeMessageFromKafka")
@consumeMessageFromKafka.dapr_binding_trigger(arg_name="triggerData", binding_name="%KafkaBindingName%")
def main(triggerData: str) -> None:
    logging.info('Python function processed a ConsumeMessageFromKafka request from the Dapr Runtime.')
    logging.info('Trigger data: ' + triggerData)